source/slang/slang-check-overload.cpp

```diff
@@ -1348,6 +1348,13 @@ namespace Slang
                 return;
             }
         }
+        // If this function is marked as 'extern' and in the overload candidate,
+        // it indicates that there is a 'export' function in this module that will
+        // overload this function, so we don't add this function to the overload candidate.
+        if (funcDecl->hasModifier<ExternModifier>())
```

Comment:
We should be adding the function to a candidate.
We need to extend `CompareOverloadCandidates` to favor `export` function instead of the `extern` function.